lib/wasm-micro-runtime-WAMR-1.1.1/core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -2756,7 +2756,7 @@ wasmtime_ssp_poll_oneoff(
                             __WASI_EVENT_FD_READWRITE_HANGUP,
                     };
                 }
-                else if ((pfds[i].revents & (POLLRDNORM | POLLWRNORM)) != 0) {
+                else if ((pfds[i].revents & (POLLIN | POLLOUT)) != 0) {
```

Comment:
Could you send your patch into wasm-micro-runtime's upstream?
https://github.com/bytecodealliance/wasm-micro-runtime
We wanted to apply your changes there instead of patching here.

Reply:
See the initial message of this PR.
https://github.com/bytecodealliance/wasm-micro-runtime/pull/2069

It's already applied upstream. So I guess it's safe to apply it here already.

Reply:
Oh, sorry. I've overlooked that message. looks reasonable. :+1: